app/code/Magento/QuoteGraphQl/Model/Resolver/MergeCarts.php

```diff
@@ -31,44 +37,93 @@ class MergeCarts implements ResolverInterface
      */
     private $cartRepository;
 
+    /**
+     * @var CustomerCartResolver
+     */
+    private $customerCartResolver;
+
+    /**
+     * @var QuoteIdToMaskedQuoteIdInterface
+     */
+    private $quoteIdToMaskedQuoteId;
+
     /**
      * @param GetCartForUser $getCartForUser
      * @param CartRepositoryInterface $cartRepository
+     * @param CustomerCartResolver $customerCartResolver
+     * @param QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
      */
     public function __construct(
         GetCartForUser $getCartForUser,
-        CartRepositoryInterface $cartRepository
+        CartRepositoryInterface $cartRepository,
+        CustomerCartResolver $customerCartResolver,
+        QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
     ) {
         $this->getCartForUser = $getCartForUser;
         $this->cartRepository = $cartRepository;
+        $this->customerCartResolver = $customerCartResolver;
+        $this->quoteIdToMaskedQuoteId = $quoteIdToMaskedQuoteId;
     }
 
     /**
      * @inheritdoc
      */
-    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
-    {
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
         if (empty($args['source_cart_id'])) {
-            throw new GraphQlInputException(__('Required parameter "source_cart_id" is missing'));
-        }
-
-        if (empty($args['destination_cart_id'])) {
-            throw new GraphQlInputException(__('Required parameter "destination_cart_id" is missing'));
+            throw new GraphQlInputException(__(
+                'Required parameter "source_cart_id" is missing'
+            ));
         }
 
         /** @var ContextInterface $context */
         if (false === $context->getExtensionAttributes()->getIsCustomer()) {
-            throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));
+            throw new GraphQlAuthorizationException(__(
+                'The current customer isn\'t authorized.'
+            ));
+        }
+        $currentUserId = $context->getUserId();
+
+        if (!isset($args['destination_cart_id'])) {
+            try {
+                $cart = $this->customerCartResolver->resolve($currentUserId);
+            } catch (CouldNotSaveException $exception) {
+                throw new GraphQlNoSuchEntityException(
+                    __('Could not create empty cart for customer'),
+                    $exception
+                );
+            }
+            $customerMaskedCartId = $this->quoteIdToMaskedQuoteId->execute(
+                (int) $cart->getId()
+            );
+        } else {
+            if (empty($args['destination_cart_id'])) {
```

Comment:
Looks like we already have this check at the very top of the resolver. I believe we may optimize these conditions.

Reply:
Hello @rogyar
Thank you for your review.
[The block of code](https://github.com/andrewbess/magento2/blob/CE-30469/app/code/Magento/QuoteGraphQl/Model/Resolver/MergeCarts.php#L92-L110) is needed for checking the parameter `destination_cart_id`.
We check if the parameter is missing then get the customer cart id from our logic, but this parameter cannot be empty. So, we throw an exception about it.

Reply:
My bad, wrong line of code reviewed previously